OrbitalTargeting/Program.cs

```diff
@@ -354,6 +361,43 @@ public void updateTargetPositionFromCustomData()
                 string[] parts = data.Split(' ');
                 _targetPosition = new Vector3(float.Parse(parts[0]), float.Parse(parts[1]), float.Parse(parts[2]));
             }
+            
+            public void updateTargetPositionFromRangeFinding()
+            {
+            // Update the targetting position by interfacing with Alastor's code
+            // 
+            // The data should be contained in the text pannel named "CruiseTargettingData"
+            // It is a series of lines of GPS position followed by a distance in km. 
+            // Spurious white space characters are ignored. 
+            // 
+            // Example : 
+            // #1:1087630:131373:1650670:#000000:19.90
+            // #2:1111172:130967:1631037:#000000:20.10
+            // #3:1101042:131107:1648341:#000000:19.94
+            // #4:1105568:145568:1631072:#000000:20.50
+            // #5:1112285:131146:1652285:#000000:30.00
+            // #6:1091072:131072:1648572:#000000:17.50
+            // #7:1109451:134313:1620297:#000000:21.55
+            // 
+            // The target is calculated by this script when launching with action "calculate_target"
+            
+                Trilateration t = new Trilateration();
+                if (_textPanelsTargetting.Count == 0)
+                {
+                    log("No targetting data text panel detected. ");
+                    return;
+                }
+                string text = _textPanelsTargetting.GetText();
```

Comment:
_textPanelsTargetting is a List<IMyTextPanel> but GetText is a method on IMyTextPanel

It is a List because Program.GridTerminalSystem.GetBlocksOfType will find all blocks matching the criteria and there could be more than one. Most likely there will be only one but it's easier to just handle it than to add an error condition.

To handle it, we can just add the text from each panel in the list to the trilateration

```suggestion
                foreach (IMyTextPanel panel in _textPanelsTargetting)
                {
                    string text = panel.GetText();
                    t.Add(text);
                }
```

Reply:
Very good point